lib/board.rb

```diff
@@ -18,10 +19,50 @@ def create_layout
     end
   end
 
+  # an array of strings to iterate through for "xxxx" or "oooo"
+  def possible_wins
+    [
+    row_1,
+    row_2,
+    row_3,
+    row_4,
+    row_5,
+    row_6,
+    column_A,
+    column_B,
+    column_C,
+    column_D,
+    column_E,
+    column_F,
+    column_G,
+    diag_1,
+    diag_2,
+    diag_3,
+    diag_4,
+    diag_5,
+    diag_6,
+    diag_7,
+    diag_8,
+    diag_9,
+    diag_10,
+    diag_11,
+    diag_12
+  ]
+  end
+
+  def winner?
+    possible_wins.each do |line|
+     return true if line.include?("XXXX") || line.include?("OOOO")
```

Comment:
my thought was to add an argument to the winner? method that calls whichever player token just finished a turn to slightly reduce the amount of checking done, but that should be an easy update